src/app/services/tests/GameService.spec.ts

```diff
@@ -316,22 +337,62 @@ describe('GameService', () => {
                 scorePlayerZero: 5,
                 scorePlayerOne: 0,
             };
-            expect(partDao.update).toHaveBeenCalledWith('partId', expectedUpdate);
+            expect(partDAO.updateAndBumpIndex).toHaveBeenCalledOnceWith('partId', Player.ONE, 4, expectedUpdate);
         }));
         it('should include the draw notification if requested', fakeAsync(async() => {
             // when updating the board to notify of a draw
-            await service.updateDBBoard('partId', MOVE_2, [0, 0], undefined, true);
+            await service.updateDBBoard('partId', Player.ONE, MOVE_2, [0, 0], undefined, true);
             // then the result is set to draw in the update
             const expectedUpdate: Partial<IPart> = {
+                lastUpdate: {
+                    index: 5,
+                    player: Player.ONE.value,
+                },
                 listMoves: [MOVE_1, MOVE_2],
                 turn: 2,
                 request: null,
                 lastMoveTime: firebase.firestore.FieldValue.serverTimestamp(),
-                result: MGPResult.DRAW.value,
+                result: MGPResult.HARD_DRAW.value,
             };
-            expect(partDao.update).toHaveBeenCalledWith('partId', expectedUpdate);
+            expect(partDAO.update).toHaveBeenCalledWith('partId', expectedUpdate);
         }));
     });
+    describe('acceptDraw', () => {
+        it('should send AGREED_DRAW_BY_ONE when call as Player.ONE', () => {
+            // Given any state of service
+            spyOn(partDAO, 'update');
+
+            // When calling acceptDraw as Player.ONE
+            service.acceptDraw('joinerId', 5, Player.ONE);
+
+            // Then PartDAO should have been called with AGREED_DRAW_BY_ONE
+            expect(partDAO.update).toHaveBeenCalledOnceWith('joinerId', {
+                lastUpdate: {
+                    index: 6,
+                    player: Player.ONE.value,
+                },
+                request: null,
+                result: MGPResult.AGREED_DRAW_BY_ONE.value,
+            });
+        });
```

Comment:
peut-être combiner les deux tests en un pour éviter de la duplication ? (Genre for player of [Player.ZERO, Player.ONE] …)

Reply:
les commentaires doivent être mis à jour !